### PR TITLE
GH-4558 Mobile View: Don't close the Left Hand Sidebar after removing an open Direct Message Channel

### PR DESCRIPTION
**Pull request by fraziern**
#### Summary
Mobile View: Don't close the Left Hand Sidebar after removing an open Direct Message Channel

#### Ticket Link
#4558 

#### Checklist
- [x] Has UI changes

### CODE REVIEW

**Comment by yangchen1** [comment]
+1 Great work @fraziern :)

**Comment by esethna** [comment]
Great work @fraziern, tested this in Chrome browser mobile view and in the iOS app. I've also tested to ensure this works for DMs outside the team.

____

FYI I've found an unrelated issue, [ticket here](https://mattermost.atlassian.net/browse/PLT-5025), where you will very breifly scroll to the top of the channel when switching channels. This is more noticable on mobile but also happens in desktop browsers.

Repro: 
1) Load two channels with content so the channel intro is not visible.
2) Switch between the channels.

Observed:
After switching, the channel intro briefly appears before you are put to the bottom of the channel (see GIF - the framerate is not optimal but you can observe the behaviour in the first switch).

scroll flash channel switch

Expected: 
You are immediately dropped at the bottom of the channel after switching.

File: webapp/components/sidebar.jsx [hunk 1]
```diff
@@ -149,11 +150,21 @@ export default class Sidebar extends React.Component {
 
         // close the LHS on mobile when you change channels
         if (this.state.activeId !== prevState.activeId) {
-            $('.app__body .inner-wrap').removeClass('move--right');
-            $('.app__body .sidebar--left').removeClass('move--right');
+            if (this.state.closedDirectChannel) {
+                this.onUpdate(() => {
+                    this.setState({closedDirectChannel: false});
+                });
```
Comment (jwilander) [inline webapp/components/sidebar.jsx]: Really we should not be using jQuery here to control whether the sidebar is open or not, it should be controlled by the state/props and the render method. That's a bigger refactor than I think is worth it for this ticket, though.

You're actually not using `this.state.closedDirectChannel` to affect rendering at all though so it really shouldn't be in state. Moving it to just be a field on the component itself, such as `this.closedDirectChannel`, would make more sense. It should also get around the hack of having to use this `onUpdate` function

Comment (fraziern) [inline webapp/components/sidebar.jsx]: @jwilander new commit tested & added.

File: webapp/components/sidebar.jsx [hunk 2]
```diff
@@ -149,11 +151,19 @@ export default class Sidebar extends React.Component {
...
+            if (this.closedDirectChannel) {
+                this.closedDirectChannel = false;
+            } else {
+                $('.app__body .inner-wrap').removeClass('move--right');
+                $('.app__body .sidebar--left').removeClass('move--right');
+            }
         }
     }
 
+    onUpdate = (cb) => {
```
Comment (jwilander) [inline webapp/components/sidebar.jsx]: Can you remove this as well? I don't see it being used anywhre

Comment (fraziern) [inline webapp/components/sidebar.jsx]: oops, yep done.

**Comment by jwilander** [comment]
Much thanks @fraziern ! Looks good

**Comment by coreyhulen** [comment]
@fraziern looks like you have merge conflicts, can you please rebase.

**Comment by fraziern** [comment]
@coreyhulen Yep, done.

**Pull request merged**